### PR TITLE
Move doxygen to GitHub Actions

### PR DESCRIPTION
**Pull request by seladb**
- Move doxygen to GitHub Actions and remove it from Cirrus CI
- Temporary comment-out all Cirrus CI tasks so we can merge PRs

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -15,684 +15,725 @@ permissions:
   contents: read
 
 jobs:
-  pre-commit:
+  doxygen:
     runs-on: ubuntu-latest
-    container: seladb/alpine317
+    container: seladb/ubuntu1804:latest
```
Comment (tigercosmos) [inline .github/workflows/build_and_test.yml:20]: @seladb why ubuntu18? we should use the lastest LTS.

Comment (seladb) [reply .github/workflows/build_and_test.yml]: We need doxygen to be installed. We installed it in all of our Ubuntu images: https://github.com/seladb/PcapPlusPlus-DockerImages/blob/master/Dockerfile-ubuntu2204

I updated the image to Ubuntu 22.04

**Comment by clementperon** [comment]
@seladb should It be a different workflow?

**Comment by seladb** [comment]
> @seladb should It be a different workflow?

I'm not sure it's necessary, because a major part is to test doxygen which is part of `build_and_test`

**Comment by tigercosmos** [comment]
> > @seladb should It be a different workflow?
> 
> I'm not sure it's necessary, because a major part is to test doxygen which is part of `build_and_test`

@seladb  I think it is worth doing that, since we can easily track the CI category at the panel:
image

**Comment by clementperon** [comment]
> > > @seladb should It be a different workflow?
> > 
> > 
> > I'm not sure it's necessary, because a major part is to test doxygen which is part of `build_and_test`
> 
> @seladb I think it is worth doing that, since we can easily track the CI category at the panel:

@seladb @tigercosmos to me the other reason is that the triggering mode could differs from the build_package but maybe i'm wrong, So it don't have a strong opinion.

File: .cirrus.yml [hunk 2]
```diff
@@ -2,97 +2,68 @@ env:
   BUILD_DIR: "build"
   GCOVR_FLAGS: "--gcov-ignore-parse-errors --exclude-throw-branches --filter Common --filter Pcap --filter Packet --xml"
```
Comment (clementperon) [inline .cirrus.yml:3]: maybe we could remove cirrus file no ?

**Review by clementperon — APPROVED** [review]

File: .cirrus.yml (hunk 2 above)
Comment (seladb) [reply .cirrus.yml:3]: I think we can remove it after we migrate all jobs to GitHub Actions. In each PR I'll remove the jobs that are migrated

**Pull request merged**